### DISCUSSION TITLE
Crop water aeration stress -- the role of LagAer

### DISCUSSION THREAD

**Discussion by WMLKalthof** [aquacropos/aquacrop#167] [Q&A]
Hi all! I am looking to implement aeration stress in an ABM. As its a model about (amongst others) farmer crop choices, distinction between crops under water stresses are important. For most crops, you have the crop_aer factor which differs, however, only rice paddy has a LagAer factor under crop_params.py. In the model itself this is used to determine which calculation of the aeration stress is used: 
```python
        if NewCond_AerDays < Crop_LagAer:
            stress = 1 - ((thRZ.S - thRZ.Act) / (thRZ.S - thRZ.Aer))
            Ksa_Aer = 1 - ((NewCond_AerDays / 3) * stress)
        elif NewCond_AerDays >= Crop_LagAer:
            Ksa_Aer = (thRZ.S - thRZ.Act) / (thRZ.S - thRZ.Aer)
``` 
Now my questions: 

1. Are these values only known for paddy rice or can I find these somewhere else? 
2. If not: should I assume that always the calculation of the elif condition should be used?
3. In an excel mock of the calculation, the first condition calculation returns negative values if NewCond_AerDays > 4. Furthermore, below 4 it returns greater stresses compared to the second condition, even though you would expect that this would be reversed, since I assume the lag days before aeration stress indicate better water logging tolerance. The mock values I used are:

| Header | Header |
|--------|--------|
| thRZ.S| 1 |
| thRZ.Act | 0.9 |
| thrz.AER | 0.7 |
| AerDays | 4 |
| 1st condition calculation | 0.11 | 
| 2nd condition calculation | 0.33|

For now I will use the second condition calculation (which I believe is also the one who was used in version 6 and below), but could you perhaps help me better understand the first? 

Thank you in advance!

**Comment by WMLKalthof** [comment]
The second part has been solved, but I am wondering whether I can find the LagAer coefficients somewhere!

**Comment by Paloschi** [comment]
### Understanding LagAer in AquaCrop-OSPy
1. Are these values only known for paddy rice or can I find these somewhere else?
In AquaCrop-OSPy, the LagAer parameter is specifically designed for paddy rice due to its unique tolerance to waterlogged conditions. Other crops typically do not have this parameter because they do not generally grow in flooded conditions like rice. You might not find LagAer values for other crops because AquaCrop-OSPy does not include them in its default crop parameters.

2. If not: should I assume that always the calculation of the elif condition should be used?
Yes, if no LagAer values are available for other crops, you should use the elif condition calculation for aeration stress. This simplification is practical and aligns with how AquaCrop handles non-rice crops.

3. Excel Mock Calculation Clarification
Your calculations and understanding seem correct. Here is the precise breakdown of how these calculations work in AquaCrop-OSPy:

- First Condition (NewCond_AerDays < Crop_LagAer):

```
stress = 1 - ((thRZ.S - thRZ.Act) / (thRZ.S - thRZ.Aer))
Ksa_Aer = 1 - ((NewCond_AerDays / 3) * stress)
```
This calculation includes a dynamic stress factor that decreases Ksa_Aer as NewCond_AerDays increases, up to a maximum of 3 days.

- Second Condition (NewCond_AerDays >= Crop_LagAer):

```
Ksa_Aer = (thRZ.S - thRZ.Act) / (thRZ.S - thRZ.Aer)
```
This simpler calculation directly computes the stress based on soil moisture content when NewCond_AerDays exceeds Crop_LagAer.

### Finding LagAer Coefficients
For crops other than paddy rice, LagAer coefficients are generally not defined in AquaCrop. You can refer to AquaCrop's comprehensive documentation or relevant agronomic studies for specific crop parameters. Typically, the default AquaCrop parameters for various crops will not include LagAer except for paddy rice.

### Summary

- The LagAer parameter is specific to paddy rice in AquaCrop-OSPy.
- For other crops, use the second condition for aeration stress calculation.
- Refer to AquaCrop's documentation or agronomic literature for specific crop parameters.

If you have specific crops in mind or need detailed help with AquaCrop-OSPy, feel free to ask!

**Reply by WMLKalthof** [reply]
Hi Paloschi,

Thank you for your swift and detailed reply! I have some additional questions based on the full calculation of the aeration stress. In the code to determine the aeration stress threshold, or thRZ_Aer, the following calculation is made:

```python
WrAer = WrAer + round(factor * 1000 * (prof.th_s[ii] - (Crop_Aer / 100)) * prof.dz[ii], 2)
``` 
For a single layer, keeping it in m as that is that is the unit used in the model I work on (ABM connected to CWatM), this code becomes the following:

```python
WrAer = (ws - (crop_aer / 100))
``` 
Where ws is the max storage capacity per layer. The factor was removed as this is accounted for later (and this also became strongly negative if the root depth was small but the total layer size large). 

This works fine for the second and third soil layer (0.95 m and 2 m, respectively), however, the upper soil layer of  my model is 0.05 m. With  Crop_Aer being >= 5 (which is the case for all crops except paddy rice), this means that the aeration stress threshold always defaults to 0.0 m or becomes negative, thus saying that there is always stress. Have I made a mistake somewhere or is it not possible to use this implementation with a layer of 0.05 m ? Do you perhaps have suggestions on how to change this factor? 

Additional question: this factor is a factor that is subtracted from the soil layer saturation water content, not multiplied / divided / etc. Thus I would assume in this calculation that smaller layers always are more sensitive to water stress. Is this correct? And if so, is this partly based on that in most crop simulation models the upper layer is smallest?

**Reply by Paloschi** [reply]
Hi WMLKalthof, you're welcome!

You're tackling some complex dynamics with soil layer interactions and aeration stress thresholds in your model. Let's break down your calculations and questions to ensure your model behaves as intended:

1. Understanding the Calculation for WrAer:

- The formula, WrAer = (ws - (crop_aer / 100)), calculates the water content at which aeration stress begins. Here, ws is the maximum water storage capacity of the soil layer, and crop_aer represents a percentage reduction from this saturation point to define the threshold for aeration stress.

- The adjustment by removing the factor is logical, especially if it led to inconsistencies or negative values when applying it to different soil layer sizes.

2. Issues with Small Soil Layers:

- For the upper soil layer with a depth of 0.05 m, the phenomenon you're observing where WrAer becomes zero or negative is indeed a practical issue. This implies that the formula in its current form may not be suitable for very shallow layers if crop_aer values are relatively high.

- The negative or zero threshold suggests that the formula assumes aeration stress under almost all conditions, which might not be realistic or intended.

3. Suggestions for Adjusting the Formula:

- One approach to address this is to incorporate a minimum threshold condition that ensures WrAer does not fall below a certain positive value, regardless of the layer depth. For example, setting a minimum viable WrAer can prevent the model from predicting unrealistic perpetual stress conditions in shallow layers.

- Alternatively, you could adjust the way crop_aer is applied, possibly scaling it based on the depth of the soil layer or revising how it impacts the threshold calculation, to reflect more nuanced soil-water interactions.

4. Sensitivity of Smaller Layers:

- Your observation that smaller layers are more sensitive to water stress under this formulation is accurate. This is generally realistic because smaller layers can quickly become saturated or dry out, impacting their ability to buffer changes in water content.

- In many crop simulation models, the uppermost layer is indeed often the smallest because it experiences the most rapid changes in moisture due to evaporation and precipitation, making its accurate modeling crucial for overall water stress dynamics.

5. Model Adjustments for Realism:

      To make this aspect of your model more realistic, consider:

- Implementing layer-specific modifications to the crop_aer application, possibly using empirical data or literature to guide how these adjustments should be scaled based on layer depth.

- Experimenting with a non-linear application of crop_aer, where the effect diminishes or increases based on specific soil properties or crop characteristics, could also help tailor the model more closely to observed behaviors.

These changes can help ensure that your model not only provides a more accurate representation of soil moisture dynamics across different soil layers but also reflects the actual susceptibility of these layers to aeration stress in a way that aligns with agronomic principles and empirical observations.
